src/lib.rs

```diff
@@ -76,11 +107,18 @@ impl From<&(SocketAddr, Option<SocketAddr>)> for SocketAddrs {
 /// This function returns an error if the local interface MTU cannot be determined.
 pub fn interface_and_mtu<A>(addrs: A) -> Result<(String, usize), Error>
 where
-    SocketAddrs: From<A>,
+    SocketAddrs: TryFrom<A>,
 {
-    let addrs = SocketAddrs::from(addrs);
+    let addrs = match SocketAddrs::try_from(addrs) {
+        Ok(addrs) => addrs,
+        Err(_e) => return Err(Error::new(ErrorKind::InvalidInput, "No address found")),
```

Comment:
@mxinden another Rust question. I thought I could do
```rust
        Err(e) => return e,
```
here, but I get this error:
```
error[E0308]: mismatched types
   --> src/lib.rs:115:26
    |
109 | pub fn interface_and_mtu<A>(addrs: A) -> Result<(String, usize), Error>
    |                                          ------------------------------ expected `std::result::Result<(std::string::String, usize), std::io::Error>` because of return type
...
115 |         Err(e) => return e,
    |                          ^ expected `Result<(String, usize), Error>`, found associated type
    |
    = note:         expected enum `std::result::Result<(std::string::String, usize), std::io::Error>`
            found associated type `<SocketAddrs as std::convert::TryFrom<A>>::Error`
    = help: consider constraining the associated type `<SocketAddrs as std::convert::TryFrom<A>>::Error` to `std::result::Result<(std::string::String, usize), std::io::Error>`
    = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
```
Again, any ideas?